plasmapy/utils/data/downloader.py

```diff
@@ -4,74 +4,409 @@
+                f"{self._API_BASE_URL}. "
+                f"Response content: {reply.content}. Exception: {err}"
             )
+            self._validate = False
+            return None
+
+        for item in info:
+            try:
+                filename = item["name"]
+                repo_sha = item["sha"]
+                download_url = item["download_url"]
+
+            # Not tested, since any URL on the gituhb API that doesn't return a 404
+            # should be a JSON with these keys
+            except (KeyError, TypeError) as err:  # coverage: ignore
+                warnings.warn(
+                    f"URL {self._API_BASE_URL} returned JSON file, "
+                    "missing expected keys 'sha' and 'download_url`."
+                    f" JSON contents: {info}. Exception: {err}"
+                )
+                filename = None
+                repo_sha = None
+                download_url = None
+
+            if filename is not None:
+                self._update_blob_entry(
+                    filename, repo_sha=repo_sha, download_url=download_url
+                )
+
+        # Save the current epoch time in the blob file as a record of when
+        # it was updated
+        self._blob_dict["_timestamp"] = time.time()
+
+        # At the end, write back to the blobfile
+        self._write_blobfile()
+
+        # The blob file has been updated: set this flag so we won't do it
+        # again on this instance of Downloader
+        self._updated_blob_file_from_repo = True
+
+    def _update_blob_entry(
+        self,
+        filename: str,
+        local_sha: str | None = None,
+        repo_sha: str | None = None,
+        download_url: str | None = None,
+    ) -> None:
+        """
+        Update an entry in the blobfile, or create a new one if one doesn't
+        exist.
+        """
+
+        if filename in self._blob_dict:
+            if local_sha is not None:
+                self._blob_dict[filename]["local_sha"] = local_sha
+            if repo_sha is not None:
+                self._blob_dict[filename]["repo_sha"] = repo_sha
+            if download_url is not None:
+                self._blob_dict[filename]["download_url"] = download_url
+        else:
+            self._blob_dict[filename] = {
+                "local_sha": local_sha,
+                "repo_sha": repo_sha,
+                "download_url": download_url,
+            }
+
+    def _http_request(self, url: str) -> requests.Response:
+        """
+        Issue an HTTP request to the specified URL, handling exceptions.
+        """
+
+        try:
+            reply = requests.get(url)  # noqa: S113
+
+        # No test coverage for this exception since we can't test it without
+        # severing the network connectivity in pytest
+        except requests.ConnectionError as err:  # coverage: ignore
+            raise requests.ConnectionError(
+                "Unable to connect to data " f"repository {self._API_BASE_URL}"
```

Comment:
```suggestion
                f"Unable to connect to data repository {self._API_BASE_URL}"
```
I suspect these were originally on two lines, and then were put on one line by ruff-format.  I think we have it set so that this is usually autofixed, but maybe that was skipped since one was an f-string and the other wasn't?